Sources/Sentry/SentryUIPerformanceTracker.m

```diff
@@ -0,0 +1,258 @@
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wshadow"
+
+    SEL selector = NSSelectorFromString(@"viewWillAppear:");
+    SentrySwizzleInstanceMethod(class, selector, SentrySWReturnType(void),
+        SentrySWArguments(BOOL animated), SentrySWReplacement({
+            NSString *spanId
+                = objc_getAssociatedObject(self, &SENTRY_UI_PERFORMANCE_TRACKER_SPAN_ID);
+
+            if (spanId == nil) {
+                SentrySWCallOriginal(animated);
+            } else {
+                [SentryPerformanceTracker.shared pushActiveSpan:spanId];
+                SentrySWCallOriginal(animated);
+                [SentryPerformanceTracker.shared popActiveSpan];
+            }
+        }),
+        SentrySwizzleModeOncePerClassAndSuperclasses, (void *)selector);
+#    pragma clang diagnostic pop
+#endif
+}
+
++ (void)swizzleViewDidAppear:(Class)class
+{
+#if SENTRY_HAS_UIKIT
+    // SentrySwizzleInstanceMethod declaration shadows a local variable. The swizzling is working
+    // fine and we accept this warning.
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wshadow"
+
+    SEL selector = NSSelectorFromString(@"viewDidAppear:");
+    SentrySwizzleInstanceMethod(class, selector, SentrySWReturnType(void),
+        SentrySWArguments(BOOL animated), SentrySWReplacement({
+            NSString *spanId
+                = objc_getAssociatedObject(self, &SENTRY_UI_PERFORMANCE_TRACKER_SPAN_ID);
+
+            if (spanId == nil) {
+                SentrySWCallOriginal(animated);
+            } else {
+                [SentryPerformanceTracker.shared pushActiveSpan:spanId];
+                SentrySWCallOriginal(animated);
+                [SentryPerformanceTracker.shared popActiveSpan];
+                [SentryPerformanceTracker.shared finishSpan:spanId];
+            }
+        }),
+        SentrySwizzleModeOncePerClassAndSuperclasses, (void *)selector);
+#    pragma clang diagnostic pop
+#endif
+}
+
++ (void)swizzleViewLayoutSubViews:(Class)class
+{
+#if SENTRY_HAS_UIKIT
+    // SentrySwizzleInstanceMethod declaration shadows a local variable. The swizzling is working
+    // fine and we accept this warning.
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wshadow"
+
+    SEL willSelector = NSSelectorFromString(@"viewWillLayoutSubviews");
+    SentrySwizzleInstanceMethod(class, willSelector, SentrySWReturnType(void), SentrySWArguments(),
+        SentrySWReplacement({
+            NSString *spanId
+                = objc_getAssociatedObject(self, &SENTRY_UI_PERFORMANCE_TRACKER_SPAN_ID);
+
+            if (spanId == nil || ![SentryPerformanceTracker.shared isSpanAlive:spanId]) {
+                SentrySWCallOriginal();
+            } else {
+                [SentryPerformanceTracker.shared pushActiveSpan:spanId];
+                NSString *layoutSubViewId =
+                    [SentryPerformanceTracker.shared startSpanWithName:@"layoutSubViews"
```

Comment:
can you explain how that works? maybe we can come up with a generic name that also makes sense for other SDKs, eg Android

Reply:
This is a UIKit process to re-organize uiview in the screen every time a layout change happens.